### PR TITLE
REENG-14 fixed decodeTime in Mapper

### PR DESCRIPTION
**Pull request by aliaksandr-lishko-paysera**
- Added validation in decodeTime method in Mapper
- Added validation covered with tests

### CODE REVIEW

**Comment by aliaksandr-lishko** [comment]
@michalblocinski please have a look

**Comment by PianistaMichal** [comment]
approve from my side

File: tests/Paysera/WalletApi/MapperTest.php [hunk 1]
```diff
@@ -1955,6 +1955,85 @@ public function testEncodeLocationEmptySpot() {
...
+        return [
+            'valid data' => [
+                'data' => '9:0',
+                'expectedException' => null,
+                'expectedResult' => new Paysera_WalletApi_Entity_Time(9, 0),
+            ],
+            'wrong format: missed minutes' => [
+                'data' => '9',
+                'expectedException' => InvalidArgumentException::class,
+                'expectedResult' => null
```
Comment (turnsturntra) [inline tests/Paysera/WalletApi/MapperTest.php:1987]: Here and other places https://github.com/paysera/php-style-guide?tab=readme-ov-file#commas-in-arrays

File: src/Paysera/WalletApi/Mapper.php [hunk 2]
```diff
@@ -1888,6 +1888,9 @@ public function encodeTime(Paysera_WalletApi_Entity_Time $object)
     public function decodeTime($input)
     {
         $time = explode(':', $input);
+        if (count($time) !== 2) {
+            throw new InvalidArgumentException('Provided time is not in valid format. Valid format is HH:MM');
```
Comment (turnsturntra) [inline src/Paysera/WalletApi/Mapper.php:1892]:
```suggestion
            throw new InvalidArgumentException('Provided time is not in valid format. Valid format is HH:MM.');
```

**Review by turnsturntra — APPROVED** [review]

File: tests/Paysera/WalletApi/MapperTest.php [hunk 3]
```diff
@@ -1955,6 +1955,85 @@ public function testEncodeLocationEmptySpot() {
...
+        if ($expectedException !== null) {
+            $this->setExpectedException($expectedException);
+        }
+
+        $result = $this->mapper->decodeTime($data);
+
+        if ($expectedResult !== null) {
+            $this->assertInstanceOf(Paysera_WalletApi_Entity_Time::class, $result);
+            $this->assertEquals($expectedResult->getHours(), $result->getHours());
+            $this->assertEquals($expectedResult->getHours(), $result->getHours());
```
Comment (turnsturntra) [inline tests/Paysera/WalletApi/MapperTest.php:1972]: Shouldn't be here minutes? :)

**Pull request merged**